File: golang/primos.go
package main

import (
	"math"
)

func primosLinear(numero int) []int {
	if numero < 2 {
		return nil
	}

	SliceDePrimos := []int{}
	
	for i := 2; i <= numero; i++ {
		ehPrimo := true
		limite := int(math.Sqrt(float64(i)))
		
		for j := 2; j <= limite; j++ {
			if i%j == 0 {
				ehPrimo = false
				break
			}
		}
		if ehPrimo {
			SliceDePrimos = append(SliceDePrimos, i)
		}
	}
	return SliceDePrimos
	
}

File: python/primos.py
import math


def primos(numero: int) -> list:
    if numero < 2:
        return []

    primos = []
    for i in range(2, numero+1):
        ehPrimo = True
        limite = int(math.sqrt(i))
        for j in range(2, limite+1):
            if i % j == 0:
                ehPrimo = False
                break

        if ehPrimo:
            primos.append(i)

    return primos
